### PR TITLE
Fix problems with nim check

### PR DESCRIPTION
**Pull request by baabelfish**
- Multi file errors are not shown in the same buffer
- Fixes parsing of error type that contain ':'

### CODE REVIEW

File: ale_linters/nim/nimcheck.vim [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 
 function! ale_linters#nim#nimcheck#Handle(buffer, lines) abort
+    let l:buffer_filename = fnameescape(fnamemodify(bufname(a:buffer), ':p:t'))
```
Comment (w0rp) [inline ale_linters/nim/nimcheck.vim]: I think you can do away with `fnameescape` here and below. You only need to escape filenames if they are being used in a command line string.

**Comment by w0rp** [comment]
Cheers! :beers:

**Comment by nkala** [comment]
:beers:

**Comment by baabelfish** [comment]
:beers:

**Pull request merged**